build-system/common/ci.js

```diff
@@ -194,6 +194,15 @@ function ciJobUrl() {
     : '';
 }
 
+/**
+ * Returns the merge commits SHA when running a CI Pull Request build.
+ * @return {string}
+ */
+function ciMergeSha() {
+  // CIRCLE_MERGE_SHA is populated by .circleci/fetch_merge_commit.sh.
+  return isCircleci ? env('CIRCLE_MERGE_SHA') : '';
+}
```

Comment:
The thumb-rule for every function in this file is that it should be usable on GH actions if ever required, and that it is based on well-known env vars from each platform.

https://github.com/ampproject/amphtml/blob/c1212924d1f98052cea36b1a5606e3550a1a2cf5/build-system/common/ci.js#L18-L24

For `CIRCLE_MERGE_SHA`, I wonder if we should just add a util function within `bundle-size/index.js`.

In addition, we'll need to use this as the baseline for visual diffs too.

https://github.com/ampproject/amphtml/blob/c1212924d1f98052cea36b1a5606e3550a1a2cf5/build-system/tasks/visual-diff/index.js#L136-L150

Reply:
SG. Let's just rename this function to `circleciMergeSha` to clearly indicate that it's not a generic CI function.

Reply:
According to https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request, the `GITHUB_SHA` is actually the merge commit! And we're mixing up the CCI SHA (which is the HEAD SHA) with the GH SHA (which is the merge SHA) in `ciCommitSha` above.

I'll rename this for now, and leave it to you to cleanup the env vars.

Reply:
`ciCommitSha` is only used during push builds, and according to [these docs](https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables), `GITHUB_SHA` is the correct variable.

I will admit that `ci.js` has gotten confusing due to all the individual use cases, and could do with a separate cleanup.

Other cleanup is coming up in #33215.